### ISSUE TITLE
sql: propagated filters are not used for span generation

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#12893]
Pursuant to #12617.

Noticed with:
```
root@:26257> create table t.kv(k int primary key, v int);
CREATE TABLE
root@:26257> explain(verbose) select * from t.kv  as a, t.kv as b where a.k = b.k and a.v < 3 and b.k > 3;
+-------+------+----------+-------------+--------------+----------+
| Level | Type |  Field   | Description |   Columns    | Ordering |
+-------+------+----------+-------------+--------------+----------+
|     0 | join |          |             | (k, v, k, v) |          |
|     0 |      | type     | inner       |              |          |
|     0 |      | equality | (k) = (k)   |              |          |
|     1 | scan |          |             | (k, v)       |          |
|     1 |      | table    | kv@primary  |              |          |
|     1 |      | filter   | v < 3       |              |          |
|     1 | scan |          |             | (k, v)       |          |
|     1 |      | table    | kv@primary  |              |          |
|     1 |      | filter   | k > 3       |              |          |
+-------+------+----------+-------------+--------------+----------+
(9 rows)
```

Expected instead:
```
root@:26257> explain(verbose) select * from t.kv  as a, (select * from t.kv where k > 3) as b where a.k = b.k and a.v < 3;
+-------+------+----------+-------------+--------------+-----------+
| Level | Type |  Field   | Description |   Columns    | Ordering  |
+-------+------+----------+-------------+--------------+-----------+
|     0 | join |          |             | (k, v, k, v) |           |
|     0 |      | type     | inner       |              |           |
|     0 |      | equality | (k) = (k)   |              |           |
|     1 | scan |          |             | (k, v)       |           |
|     1 |      | table    | kv@primary  |              |           |
|     1 |      | filter   | v < 3       |              |           |
|     1 | scan |          |             | (k, v)       | +k,unique |
|     1 |      | table    | kv@primary  |              |           |
|     1 |      | spans    | /4-         |              |           |
|     1 |      | filter   | k > 3       |              |           |
+-------+------+----------+-------------+--------------+-----------+
(10 rows)
```

(notice the optimized span on the 2nd scan)

**Comment by knz** [comment]
The problem is that the index selection / span generation is performed by selectIndex, which currently only works with renderNode. This logic needs to migrate away from there so as to support standalone scanNodes.

**Comment by knz** [comment]
Fixed by #13149
